src/V3LinkDot.cpp

```diff
@@ -2646,6 +2646,47 @@ class LinkDotResolveVisitor final : public VNVisitor {
         m_ds.init(m_curSymp);
         iterateNull(nodep);
     }
+    static const AstNodeDType* getExprDTypep(const AstNodeExpr* selp) {
+        auto getElemDTypep = [](const AstNodeDType* dtypep) {
+            while (true) {
+                if (const AstBracketArrayDType* const bracketArrayp
+                    = VN_CAST(dtypep, BracketArrayDType)) {
+                    dtypep = bracketArrayp->childDTypep();
+                } else if (const AstDynArrayDType* const packArrayDTypep
+                           = VN_CAST(dtypep, DynArrayDType)) {
+                    dtypep = packArrayDTypep->childDTypep();
+                } else if (const AstQueueDType* const queueDTypep = VN_CAST(dtypep, QueueDType)) {
+                    dtypep = queueDTypep->childDTypep();
+                } else {
+                    break;
+                }
+            }
+            return dtypep;
```

Comment:
Prefer just make a standard function.  Use "adtypep" for each variable name inside the `if`'s.

Probably need a `->skipRefp()`. Have some of the tests use `typedef` instead of the types directly to test this.